telluric/util/local_tile_server.py

```diff
@@ -1,60 +1,99 @@
 import asyncio
 import folium
-import rasterio
-import tornado.web
 import concurrent.futures
-
-from tornado import gen
 from threading import Thread, Lock
+from collections import namedtuple
+import tornado.web
+from tornado import gen
 from tornado.ioloop import IOLoop
-from rasterio.enums import Resampling
-from telluric.constants import WGS84_CRS
 from tornado.httpserver import HTTPServer
 from tornado.concurrent import run_on_executor
 
+import rasterio
+from rasterio.enums import Resampling
+
+import telluric as tl
+from telluric.constants import WGS84_CRS
+from telluric.constants import MERCATOR_RESOLUTION_MAPPING, WEB_MERCATOR_CRS
 
-executor = concurrent.futures.ThreadPoolExecutor(50)
+
+raster_executor = concurrent.futures.ThreadPoolExecutor(50)
+
+ServedObj = namedtuple('ServedObj', 'obj footprint')
 
 
 class TileServerHandler(tornado.web.RequestHandler):
-    _tread_pool = executor
+    _thread_pool = raster_executor
 
-    def initialize(self, rasters, resampling):
-        self.rasters = rasters
+    def initialize(self, objects, resampling):
+        self.objects = objects
         self.resampling = resampling
 
     @gen.coroutine
-    def get(self, raster_id, x, y, z):
-        png_tile = yield self._get_png_tile(int(raster_id), int(x), int(y), int(z))
-        if png_tile:
-            self.set_header("Content-type", "image/png")
-            self.finish(png_tile)
+    def get(self, object_id, x, y, z):
+        # the import is here to eliminate recursive import
+        from telluric.collections import BaseCollection
```

Comment:
why the import is here?

Reply:
as it say in the comment, to prevent recursive import through the plotting mixin